b2luigi/basf2_helper/tasks.py

```diff
@@ -46,21 +46,17 @@ def process(self):
 
         try:
             import basf2
-            import ROOT
         except ImportError:
             raise ImportError("Can not find ROOT or basf2. Can not use the basf2 task.")
 
         if self.num_processes:
             basf2.set_nprocesses(self.num_processes)
 
-        if self.max_event:
-            ROOT.Belle2.Environment.Instance().setNumberEventsOverride(self.max_event)
-
         path = self.create_path()
 
         path.add_module("Progress")
         basf2.print_path(path)
-        basf2.process(path)
+        basf2.process(path=path, max_event=self.max_event if self.max_event else 0)
```

Comment:
This is a bit nit-picky but I would move this inline if-else into it's separate line. 

```suggestion
        max_event = self.max_event if self.max_event else 0
        basf2.process(path=path, max_event=max_event)
```
Optionally even use a real if-else instead of ternary operator style:
```suggestion
        if self.max_event:
            max_event = self.max_event
        else:
            max_event else 0
        basf2.process(path=path, max_event=max_event)
```
 The reason is that I think code is more readable, yields better diffs and is easier to debug if every piece of logic is on its own line. If too many separate happen in one line it's more difficult to reason about it, though I use that style style in my own scripts sometimes. But few lines of code are not a goal in itself.
 
 From the above two suggestions, choose whichever you like more or refactor that yourself if you have an even better idea.